src/pages/YourShoppingListsHere/LiYuan/index.tsx

```diff
@@ -0,0 +1,89 @@
+import styled from 'styled-components'
+import type { InputRef } from 'antd'
+import type { ColumnType } from 'antd/es/table'
+import { Space, Button, Form, Input, Popconfirm, Table } from 'antd'
+import type { FormInstance } from 'antd/es/form'
+import React, { useContext, useEffect, useRef, useState } from 'react'
+
+// Do not edit the other parts of the code!
+
+const ShoppingListDiv = styled.div`
+  margin: 1.5rem 1rem;
+`
+
+const SL_LiYuan = () => {
+  const data = [
+    {
+      key: 0,
+      name: 'Eggs',
+    },
+    {
+      key: 1,
+      name: 'Pineapples',
+    },
+  ]
+
+  const [Data, setData] = useState(data)
+  const [value, setValue] = useState('')
+
+  const columns = [
+    {
+      key: 'name',
+      title: 'Item Name',
+      dataIndex: 'name',
+    },
+    {
+      key: 'actions',
+      title: 'Actions',
+      width: 400,
+      render: (record: any) => {
+        return (
+          <Button onClick={() => Delete(record)} type="primary" style={{ marginBottom: 16 }}>
+            Delete
+          </Button>
+        )
+      },
+    },
+  ]
+  const [count, setCount] = useState(2)
+
+  const Delete = (record: any) => {
+    setData((pre) => {
+      return pre.filter((person) => person.key != record.key)
+    })
+  }
+
+  const Add = () => {
+    const newData = {
+      key: count,
+      name: value,
+    }
+    setData((pre) => {
+      return [...Data, newData]
+    })
+    setCount((count) => {
+      return count + 1
+    })
+  }
```

Comment:
You should check if the input is empty anot cos i can add blank entries...
![image](https://user-images.githubusercontent.com/75428027/196075683-5a96dc7e-d7cb-4bce-8a71-9ed53f06e90b.png)

Reply:
Updated in commit 16ef7693a38a32227028e8df0c92edb022ef9d37